plugin/action/move/move.go

```diff
@@ -0,0 +1,224 @@
+    ...
+```
+The original event:
+```json
+{
+  "service": "test",
+  "log": {
+    "level": "error",
+    "message": "error occurred",
+    "ts": "2023-10-30T13:35:33.638720813Z",
+    "stream": "stderr"
+  },
+  "zone": "z501"
+}
+```
+The resulting event:
+```json
+{
+  "service": "test",
+  "log": {
+    "level": "error",
+    "message": "error occurred",
+    "ts": "2023-10-30T13:35:33.638720813Z"
+  },
+  "other": {
+    "stream": "stderr",
+    "zone": "z501"
+  }
+}
+```
+---
+```yaml
+pipelines:
+  example_pipeline:
+    ...
+    actions:
+    - type: move
+      mode: block
+      target: other
+      fields:
+        - log
+    ...
+```
+The original event:
+```json
+{
+  "service": "test",
+  "log": {
+    "level": "error",
+    "message": "error occurred",
+    "ts": "2023-10-30T13:35:33.638720813Z",
+    "stream": "stderr"
+  },
+  "zone": "z501",
+  "other": {
+    "user": "ivanivanov"
+  }
+}
+```
+The resulting event:
+```json
+{
+  "log": {
+    "level": "error",
+    "message": "error occurred",
+    "ts": "2023-10-30T13:35:33.638720813Z"
+  },
+  "other": {
+    "user": "ivanivanov",
+    "service": "test",
+    "zone": "z501"
+  }
+}
+```
+}*/
+
+type Plugin struct {
+	config *Config
+	fields map[string][]string
+}
+
+const (
+	modeAllow string = "allow"
+	modeBlock string = "block"
+)
+
+// ! config-params
+// ^ config-params
+type Config struct {
+	// > @3@4@5@6
+	// >
+	// > The list of the fields to move.
+	// >> In `block` mode, the maximum `fields` depth is 1.
+	Fields []cfg.FieldSelector `json:"fields" slice:"true" required:"true"` // *
+
+	// > @3@4@5@6
+	// >
+	// > The mode of the moving. Available modes are one of: `allow|block`.
+	Mode string `json:"mode" required:"true"` // *
+
+	// > @3@4@5@6
+	// >
+	// > The target field of the moving.
+	// >> If the `target` field is existing non-object field, it will be overwritten as object field.
+	// >
+	// >> In `block` mode, the maximum `target` depth is 1.
+	Target  cfg.FieldSelector `json:"target" parse:"selector" required:"true"` // *
+	Target_ []string
+}
+
+func (c *Config) validate() error {
+	if !(c.Mode == modeAllow || c.Mode == modeBlock) {
+		return fmt.Errorf("invalid mode %q", c.Mode)
+	}
+	if c.Mode == modeBlock && len(c.Target_) > 1 {
+		return errors.New(`in "block" mode, the maximum "target" depth is 1`)
+	}
+	return nil
+}
+
+func init() {
+	fd.DefaultPluginRegistry.RegisterAction(&pipeline.PluginStaticInfo{
+		Type:    "move",
+		Factory: factory,
+	})
+}
+
+func factory() (pipeline.AnyPlugin, pipeline.AnyConfig) {
+	return &Plugin{}, &Config{}
+}
+
+func (p *Plugin) Start(config pipeline.AnyConfig, params *pipeline.ActionPluginParams) {
+	p.config = config.(*Config)
+	if err := p.config.validate(); err != nil {
+		params.Logger.Fatalf("invalid config: %s", err.Error())
+	}
+
+	p.fields = make(map[string][]string)
+	isBlockMode := p.config.Mode == modeBlock
+	for _, fs := range p.config.Fields {
+		// in `block` mode, max field depth is 1
```

Comment:
let's add it to doc

Reply:
it's already in doc, similar for `target` field
![image](https://github.com/ozontech/file.d/assets/51014712/c107dc28-a97d-4333-b962-f916b39bea9f)